src/main/java/com/direwolf20/buildinggadgets/common/building/implementation/IBuildingMode.java

```diff
@@ -0,0 +1,39 @@
+package com.direwolf20.buildinggadgets.common.building.implementation;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.util.List;
+
+/**
+ * Represents a mode that can be used by some gadget,
+ */
+public interface IBuildingMode {
+
+    //TODO move to iterator system
+    List<BlockPos> computeCoordinates(World world, EntityPlayer player, BlockPos hit, EnumFacing sideHit, ItemStack tool);
```

Comment:
Ideally this would be converted to accepting a Consumer, who'd then be called on the appropiate positions. (Can do that after the PR, if you think that is beyond this PR's scope)

Reply:
Thinking a bit about it, this might cause Problems with rendering as that seems to require a certain order...?!? I don't know, we need to look into that later (if you have any ideas, feel free to suggest or implement them...)